### PR TITLE
HC-954  'Error: avrdude' when uploading

### PR DESCRIPTION
**Pull request by seaxwi**
**Changes proposed in this pull request**
Various improvements to content and structure.

### CODE REVIEW

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md [hunk 1]
```diff
@@ -2,58 +2,87 @@
...
 
-> Text following `avrdude: ser_open():` can appear in a different language if your system language is not set to English.
+You can also read about these specific errors:
 
-While these errors can be somewhat ambiguous, they are most often caused by failing or incorrectly configured hardware. It is likely that the error can be solved by following this troubleshooting procedure.
+* [avrdude: stk500_recv(): programmer is not responding
+   avrdude: stk500_getsync() attempt X of 10: not in sync](#avrdude-stk500_recv-and-stk500_getsync)
+* [avrdude: stk500v2_ReceiveMessage(): timeout
+  avrdude: stk500v2_getsync(): timeout communicating with programmer](#avrdude-stk500_recv-and-stk500_getsync)
+* [avrdude: ser_open(): can't open device "portName": no such file or directory](#avrdude-ser_open-cant-open-device-no-such-file-or-directory)
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Upload/Error-avrdude-when-uploading.md:15]: The commented out "TODO" list below needs to be implemented for the links to work

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md [hunk 2]
```diff
@@ -2,58 +2,87 @@
...
+   ![The reset button on Arduino UNO.](img/uno-reset-button.png)
 
-6. Make sure that you have closed the serial monitor.
+4. Check port and cable connections:
 
-7. Press the reset button and see if the LED blinks. If the LED doesn't blink there is a high chance that the bootloader of your board is corrupted.
+   * Try disconnecting and reconnecting the board to the computer, and confirm that the power LED (usually labeled on the board "ON") is lit. On most boards, an orange LED will also blink when the board is plugged in. <!-- TODO: Simplify -->
 
-8. Make sure you have updated and installed all the necessary drivers for the board. If you already have the drivers, try reinstalling them.
+   * If possible, connect the board directly to a USB port. Avoid using a USB hub, or try a dif
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Upload/Error-avrdude-when-uploading.md:56]:
```suggestion
   * If possible, connect the board directly to a USB port. Avoid using a USB hub.
```
I don't know if there was supposed to be more information here.

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md [hunk 3]
```diff
@@ -2,58 +2,87 @@
...
 
-    * [General overview on bootloader burning](https://www.arduino.cc/en/Tutorial/BuiltInExamples/ArduinoISP)
-    * [Using two Arduino Mega boards](https://support.arduino.cc/hc/en-us/articles/360012048060)
-    * [Using two Arduino UNO boards](https://support.arduino.cc/hc/en-us/articles/360012048080)
+6. Press the reset button and see if the LED blinks. If the LED doesn't blink there is a high chance that the bootloader of your board is corrupted.
+
+7. Try a complete uninstall (including deleting the [Arduino15 folder](https://support.arduino.cc/hc/en-us/articles/360018448279) of the Arduino IDE. Then reinstall the IDE using the latest version.
+
+8. For some boards, you can do a [loopback test](https://support.arduino.cc/hc/en-us/articles/360020366520) to test the USB to serial chip. If the loopback test passes, or if you can't do a loopback test, the bootloader may be corrupted or missing. In this case, try [burning a fresh bootloader](https://docs.arduino.cc/built-in-examples/arduino-isp/ArduinoISP). The loopback test can be done with these boards:
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Upload/Error-avrdude-when-uploading.md:68]:
```suggestion
8. For some boards, you can do a [loopback test](https://support.arduino.cc/hc/en-us/articles/360020366520) to test the USB to serial chip. If the loopback test passes, the bootloader may be corrupted or missing. In this case, try [burning a fresh bootloader](https://docs.arduino.cc/built-in-examples/arduino-isp/ArduinoISP). The loopback test can be done with these boards:
```

This information is a bit confusing. In the [updated version of "how to do a loopback test"](https://github.com/arduino/help-center-content/pull/22/files?short_path=82742e1#diff-82742e1cd70d42fd21f29977c195455fdbe17d71461f5bd98374b6f1811531a1) says that a failed loopback test indicates that the USB to TTL serial adapter is damaged. 

Maybe we should only mention the passed loopback test case.

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md [hunk 4]
```diff
@@ -63,17 +92,83 @@ avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00
...
+3. Restart your computer.
+
+4. [Find and stop process blocking a port](https://support.arduino.cc/hc/en-us/articles/4407830972050-Find-and-stop-process-blocking-a-port).
+
+---
+<a name="ser_open-permission-denied"></a>
+
+## avrdude: ser_open(): can't open device <port>: Permission denied
+
+<!-- * -->If you get this error on Linux, [make sure your user account is part of the dialout group](https://support.arduino.cc/hc/en-us/articles/360016495679).
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Upload/Error-avrdude-when-uploading.md:164]: This is not showing up as a link in rich diff view.

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md [hunk 5]
```diff
@@ -63,17 +69,111 @@ avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00
...
+This error can happen if the board on the selected port has been disconnected from the computer, or reassigned to a different port.
+
+1. Make sure the correct port is selected in _Tools > Port_.
+
+   * For step-by-step instructions, see [Select board and port in Arduino IDE](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE).
+   * If you can't find the port, see [If your board does not appear in the port menu](https://support.arduino.cc/hc/en-us/articles/4412955149586-If-your-board-does-not-appear-in-the-port-menu).
+
+2. If using the classic Nano, try selecting a different processor in _Tools > Processor_.
+
+> On Windows, you can [manually set a port for the board](https://support.arduino.cc/hc/en-us/articles/360016420140#set-COM-port) to stop it from being reassigned.
```
Comment (seaxwi) [inline content/Software and Downloads/Upload/Error-avrdude-when-uploading.md:120]: This link will work when the article is updated with df205d1c6ba6e38528890988930f2c565a90f523

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md (hunk 2 above)
Comment (seaxwi) [reply content/Software and Downloads/Upload/Error-avrdude-when-uploading.md]: 50973d56e1c6fb3498fd5c7952b66e3163155359

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md (hunk 3 above)
Comment (seaxwi) [reply content/Software and Downloads/Upload/Error-avrdude-when-uploading.md]: I've removed this from the "general checklist", as it only is relevant for a limited set of boards and errors.

But see the changes to `avrdude: stk500_recv(): programmer is not responding`.

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md (hunk 4 above)
Comment (seaxwi) [reply content/Software and Downloads/Upload/Error-avrdude-when-uploading.md]: 50973d56e1c6fb3498fd5c7952b66e3163155359

**Comment by seaxwi** [comment]
@Renat0Ribeir0

* Cleaned up comments
   * `avrdude: ser_open(): can't open device "portName": the system cannot find the file specified` will be covered in HC-963
   * Moving test cases to different documentation
* Trimmed the general checklist, and ephesised the error-specific steps in the introduction. More in-depth general troubleshooting is better for [Errors when uploading a sketch](https://support.arduino.cc/hc/en-us/articles/4403365313810-Errors-when-uploading-a-sketch) which will be updated in HC-949

File: content/Software and Downloads/Upload/Error-avrdude-when-uploading.md [hunk 6]
```diff
@@ -63,17 +69,111 @@ avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00
...
+
+   * Arduino Uno (classic)
+   * Arduino Uno Rev3
+   * Arduino Uno Rev3 SMD
+   * Arduino Mega (classic)
+   * Arduino Mega2560 Rev3
+   * Arduino Mega ADK Rev3
+   * Arduino Nano (classic)
+
+   If the loopback test passes, or if you cant do a loopback test, reprogramming the bootloader may resolve the issue. [ArduinoISP](https://docs.arduino.cc/built-in-examples/arduino-isp/ArduinoISP) can be used for boards with AtMega chips. You can also search for [articles in the Help Center](https://support.arduino.cc/hc/en-us/search?utf8=%E2%9C%93&query=bootloader).
```
Comment (Renat0Ribeir0) [inline content/Software and Downloads/Upload/Error-avrdude-when-uploading.md:90]:
```suggestion
   If the loopback test passes, or if you can't do a loopback test, reprogramming the bootloader may resolve the issue. [ArduinoISP](https://docs.arduino.cc/built-in-examples/arduino-isp/ArduinoISP) can be used for boards with AtMega chips. You can also search for [articles in the Help Center](https://support.arduino.cc/hc/en-us/search?utf8=%E2%9C%93&query=bootloader).
```

**Review by Renat0Ribeir0 — APPROVED** [review]
Just one typo to correct, otherwise LGTM 👍

**Pull request merged**